### ISSUE TITLE
kubernetes: Add config support

### ISSUE THREAD

**Issue by cemo** [hashicorp/terraform#14565]
Hi @radeksimko, the progress so far regarding kubernetes is really nice. I am trying to port a project which is using makefiles and I am trying to remove necessary makefiles as much as possible. Here is a relevant part might take your interest. It is creating a cluster by terraform and than using some shell scripts to generate a cluster kubeconfig. Some of the commands are like this:

```
kubectl config set-cluster
kubectl config set-credentials
kubectl config set-context
```
It would be nice to automate this part as well by terraform. What do you think?

**Comment by radeksimko** [comment]
Hi @cemo 
thank you for raising this suggestion.

I can see where you're coming from (automation all the way through 👍 🤖 🎉 ), but I'm not sure that Terraform is the right tool for automation this specific task - let me try to explain below why.

AFAIK the master credentials (also exposed by Terraform from GKE resource) should only be used by a small group of people, possibly just an operator. For everyone else there's `Role` & `ServiceAccount` which can limit the exposed API and therefore reduce blast radius for a given set of credentials.

The Terraform K8S provider aims to talk Kubernetes API. What we're discussing here is effectively filesystem API. As far as I'm aware you cannot get any credentials from the K8S API directly.

Generally speaking users/credentials/ACL handling sounds more like a Vault feature as Vault is better designed to handle such sensitive data.

Specifically for config management - I raised a very similar issue a while back: https://github.com/hashicorp/vault/issues/1246 - maybe it's worth mentioning this use case there too?

With all the above - do you agree we can close this issue?

**Comment by cemo** [comment]
Hi @radeksimko,

I am running on AWS. Currently there are maybe more than 10 different options to bootstrap a k8s cluster on AWS. However none of them is suitable for my taste right now. I wanted a terraform module to simply include in my projects to use. I decided to this way since I believe in terraform so much and greatness 💯  of terraform's team and expressive declarative language. I have built all cluster and its related things in terraform. There is really little ugly parts but I am satisfied with progress so far. I can bootstrap multiple clusters in a VPC. 

Since I have already providing all certificates and related things myself to cluster by means of Terraform TLS provider, there is no issue about credentials. They are already in the project. 

If you think that this is still not suitable, we can close this issue. 

Thanks again for the things you and your friend have done for the community. 🥇
